src/main/java/updater/JarUpdater.java

```diff
@@ -177,27 +185,27 @@ private void moveFileToBackup(File file) throws IOException {
      */
     private void restoreBackupOfFile(File file) throws IOException {
         log("Restoring backup");
-        String backupFilename = getBackupFilename(file.getName());
+        String backupFilename = getJarBackupFilename(file.getName());
         File backup = new File(file.getParent(), backupFilename);
         JarUpdaterFileHandler.moveFileWithReplaceExisting(backup.toPath(), file.toPath());
     }
 
     /**
-     * Removes the backup file of a given filename
-     * @param file the file which backup should be removed
-     * @throws IOException if failed to remove backup file
+     * Get backup name of jar file.
+     * If backup suffix option is provided, it will be used instead of standard backup suffix.
+     * If filename does not end with ".jar", append standard backup suffix to the filename
+     * @param filename filename which backup name should be generated
+     * @return backup filename
      */
-    private void removeBackupOfFileIfExists(File file) throws IOException {
-        log("Removing backup");
-        String backupFilename = getBackupFilename(file.getName());
-        File backup = new File(file.getParent(), backupFilename);
-        if (!Files.deleteIfExists(backup.toPath())) {
-            log(String.format("No backup file %s", file.getName()));
+    private String getJarBackupFilename(String filename) {
+        Pattern jarFilenamePattern = Pattern.compile("^(.*)\\.(jar|JAR)$");
```

Comment:
The case insensitivity can be handled with `Pattern.compile("regex", Pattern.CASE_INSENSITIVE)`.